source/extensions/quic_listeners/quiche/envoy_quic_alarm.cc

```diff
@@ -12,7 +12,15 @@ void EnvoyQuicAlarm::CancelImpl() { timer_->disableTimer(); }
 
 void EnvoyQuicAlarm::SetImpl() {
   // TODO(#7170) switch to use microseconds if it is supported.
-  timer_->enableTimer(std::chrono::milliseconds(getDurationBeforeDeadline().ToMilliseconds()));
+  quic::QuicTime::Delta duration = getDurationBeforeDeadline();
+  // Round up the duration so that any duration < 1ms will not be
+  // triggered within current event loop.
+  // QUICHE alarm has finer granularity than 1ms, and it is not expected to be
+  // scheduled in current event loop. This bit is a bummer in QUICHE, and we are
+  // working on the fix. Once upstream is fixed, we no longer need to round up
+  // the duration.
+  timer_->enableTimer(std::chrono::milliseconds(
```

Comment:
What do you mean by upstream fix? Do you mean change enableTimer to accept a microseconds instead? 
The underlying timer seems to be capable of handling smaller units of time.

Reply:
Upstream->QUICHE

Thanks for pointing out that enableTimer() supports higher resolution now. I switched to use that.